zerver/views/registration.py

```diff
@@ -137,7 +137,7 @@ def accounts_register(request):
             form = RegistrationForm(
                 initial={'full_name': hesiod_name if "@" not in hesiod_name else ""})
             name_validated = True
-        elif settings.POPULATE_PROFILE_VIA_LDAP:
+        elif settings.POPULATE_PROFILE_VIA_LDAP and not realm_creation:
```

Comment:
(Or maybe even use a different form object that's a subclass of RegistrationForm`?  Idunno what makes sense, that might be uglier)